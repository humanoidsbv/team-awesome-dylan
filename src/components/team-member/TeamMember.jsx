import React, { useState } from 'react';
import PropTypes from 'prop-types';

import MenuDownIcon from '../../assets/icons/icon-arrow-down.svg';
import styles from './TeamMember.module.css';
import teamMemberAvatar from '../../../static/pictures/picture-tnt.jpg';

const TeamMember = ({
  currentEmployer,
  employeeFunction,
  employeeNumber,
  firstName,
  lastName,
  startingDate
}) => {
  const [isTeamMemberActive, setTeamMemberActivity] = useState(false);

  const toggleTeamMemberActivity = () => {
    setTeamMemberActivity(!isTeamMemberActive);
  };

  const startDate = new Date(startingDate).toDateString();

  return (
    <div
      className={`${
        isTeamMemberActive
          ? styles.teamMemberEntryActive
          : styles.teamMemberEntry
      }`}
    >
      <div className={styles.teamMemberBasicInfo}>
        <img
          className={styles.teamMemberAvatar}
          src={teamMemberAvatar}
          alt="Logo"
        />
        <div className={styles.teamMemberTitle}>
          <span className={styles.teamMemberName}>
            {`${firstName} ${lastName}`}
          </span>
          <span className={styles.teamMemberFunction}>{employeeFunction}</span>
        </div>
        <div className={styles.extraInfoDesktop}>
          <div className={styles.teamMemberNumber}>
            <span className={styles.extraInfoMainDesktop}>
              {employeeNumber}
            </span>
            <span className={styles.extraInfoSubDesktop}>Employee number</span>
          </div>
          <div className={styles.extraEmployerInfo}>
            <span className={styles.extraInfoMainDesktop}>
              {currentEmployer}
            </span>
            <span className={styles.extraInfoSubDesktop}>Current Employer</span>
          </div>
          <div className={styles.extraStartingDate}>
            <span className={styles.extraInfoMainDesktop}>{startDate}</span>
            <span className={styles.extraInfoSubDesktop}>Starting date</span>
          </div>
        </div>
        <MenuDownIcon
          className={styles.menuDownIcon}
          onClick={toggleTeamMemberActivity}
        />
      </div>
      <div
        className={`${
          isTeamMemberActive ? styles.extraInfoActive : styles.extraInfo
        }`}
      >
        <div className={styles.extraInfoHeader}>
          <span className={styles.extraInfoHeaderText}>
            Detailed information about             {firstName}
          </span>
        </div>
        <div className={styles.extraEmployerInfo}>
          <span className={styles.extraInfoMain}>{currentEmployer}</span>
          <span className={styles.extraInfoSub}>Current Employer</span>
        </div>
        <div className={styles.extraStartingDate}>
          <span className={styles.extraInfoMain}>{startDate}</span>
          <span className={styles.extraInfoSub}>Starting date</span>
        </div>
      </div>
    </div>
  );
};

TeamMember.propTypes = {
  currentEmployer: PropTypes.string.isRequired,
  employeeFunction: PropTypes.string.isRequired,
  employeeNumber: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired,
  startingDate: PropTypes.string.isRequired
};

export default TeamMember;
