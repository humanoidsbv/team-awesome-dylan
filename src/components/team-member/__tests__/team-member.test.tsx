import * as React from 'react';
import { shallow } from 'enzyme';

import TeamMember from '../TeamMember';

it('matches snapshot', () => {
  const teamMember = shallow(
    <TeamMember
      currentClient = 'Test'
      employeeFunction = 'Design'
      employeeNumber = 'HUM_001'
      firstName = 'test'
      lastName = 'test'
      startingDate = 'test'
    />
  );
  expect(teamMember).toMatchSnapshot();
});
