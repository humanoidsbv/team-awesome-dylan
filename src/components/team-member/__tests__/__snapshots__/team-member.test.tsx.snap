// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="teamMemberEntry"
>
  <div
    className="teamMemberBasicInfo"
  >
    <img
      alt="Logo"
      className="teamMemberAvatar"
      src="test-file-stub"
    />
    <div
      className="teamMemberTitle"
    >
      <span
        className="teamMemberName"
      >
        test test
      </span>
      <span
        className="teamMemberFunction"
      >
        Design
      </span>
    </div>
    <div
      className="extraInfoDesktop"
    >
      <div
        className="teamMemberNumber"
      >
        <span
          className="extraInfoMainDesktop"
        >
          HUM_001
        </span>
        <span
          className="extraInfoSubDesktop"
        >
          Employee number
        </span>
      </div>
      <div
        className="extraClientInfo"
      >
        <span
          className="extraInfoMainDesktop"
        >
          Test
        </span>
        <span
          className="extraInfoSubDesktop"
        >
          Current Client
        </span>
      </div>
      <div
        className="extraStartingDate"
      >
        <span
          className="extraInfoMainDesktop"
        >
          Invalid Date
        </span>
        <span
          className="extraInfoSubDesktop"
        >
          Starting date
        </span>
      </div>
    </div>
    <test-file-stub
      className="menuDownIcon"
      onClick={[Function]}
    />
  </div>
  <div
    className="extraInfo"
  >
    <div
      className="extraInfoHeader"
    >
      <span
        className="extraInfoHeaderText"
      >
        Detailed information about test
      </span>
    </div>
    <div
      className="extraClientInfo"
    >
      <span
        className="extraInfoMain"
      >
        Test
      </span>
      <span
        className="extraInfoSub"
      >
        Current Client
      </span>
    </div>
    <div
      className="extraStartingDate"
    >
      <span
        className="extraInfoMain"
      >
        Invalid Date
      </span>
      <span
        className="extraInfoSub"
      >
        Starting date
      </span>
    </div>
  </div>
</div>
`;
